### PR TITLE
Remove all manual steps from kubevirtci

### PR DESCRIPTION
**Pull request by rmohr**
**Remove all unsupported providers**

* All k8s versions older than 1.17 are gone now
* All openshift versions are gone now

**Use gocli for provisioning**

Remove the bash script which still did the provisioning and further introduce a package concept which allows gocli to automatically discover all parameters based on our already established package structure.

To provision a cluster, this is now sufficient:

```
gocli provision cluster-provision/k8s/1.19
```

The resulting provisioned cluster will be named `kubevirtci/k8s-1.19`.

**Base images are now always built with the cluster**

Remove the need to publish base images. the PR jobs will now also always build the base image. The publish script will build the base images once and they will be used for all clusters which it provisions. The base image will therefore never be explicitly released but still shared by all published  clusters of a commit.

**How to vendor kubevirtci after this change**

There are two main difference to before. 
1. One is now supposed to vendor in tags, created from a postsubmit job. Vendoring a commit directly will not work anymore
1. The tag which was used to checkout the repo needs to be exported in `KUBEVIRTCI_TAG`, so that the checked out code can pull the right `gocli` container.

**Additional notes**

Until https://github.com/kubevirt/project-infra/pull/666 is merged, no longer needed jobs will fail. We only care about:

* check-provision-k8s-1.19
* check-provision-k8s-1.18
* check-provision-k8s-1.17

### CODE REVIEW

**Comment by rmohr** [comment]
Required job modifications are here: https://github.com/kubevirt/project-infra/pull/666

**Comment by rmohr** [comment]
/hold

Needs to be coordinated with https://github.com/kubevirt/project-infra/pull/666.

**Comment by rmohr** [comment]
Everything seems to work so far (also checked the publish postsubmit on my privat repos). Happy to get reviews. Once I have approval, I will coordinate the merge of the two PRs.

File: cluster-provision/gocli/docker/docker.go
Comment (qinqon) [inline cluster-provision/gocli/docker/docker.go:50]: I remember there were some changes on quota at docker.io and we may need to move to quay.io let's at least move `docker.io` to a constant.

File: cluster-provision/k8s/check-cluster-up.sh
Comment (qinqon) [inline cluster-provision/k8s/check-cluster-up.sh:6]: We can change working directory with make `make -C ../gocli container`

File: cluster-provision/k8s/provision.sh
Comment (qinqon) [inline cluster-provision/k8s/provision.sh:13]: make -C ../gocli cli

File: publish.sh
Comment (qinqon) [inline publish.sh:11]: `make -C cluster-provision/gocli container`

File: cluster-provision/gocli/cmd/provision.go
Comment (qinqon) [inline cluster-provision/gocli/cmd/provision.go:91]: We use gocli provision with one argument at `publish.sh` like `../gocli/build/cli provision ${provision_dir}` maybe we can remove this.

**Review by qinqon — CHANGES_REQUESTED** [review]
Super small nits, you can ignore all of them.

File: cluster-provision/gocli/cmd/provision.go
Comment (rmohr) [reply cluster-provision/gocli/cmd/provision.go:71]: The idea is that we stick with the version inside the file and that we bump the patch level with e.g. a periodic which then creates specific PRs with the new minor version. Very often minor-version introduced kubeadm bugs which are most of the time unrelated to what one wants to a chive in the PR.

File: cluster-provision/gocli/docker/docker.go
Comment (rmohr) [reply cluster-provision/gocli/docker/docker.go:50]: This is only here because docker.io images are treated special on the client side. The `docker.io` prefix is basically not shown when listing the images.

File: cluster-provision/gocli/cmd/provision.go
Comment (rmohr) [reply cluster-provision/gocli/cmd/provision.go:91]: Done.

File: cluster-provision/gocli/cmd/provision.go
Comment (rmohr) [reply cluster-provision/gocli/cmd/provision.go:285]: Done

File: cluster-provision/gocli/cmd/provision.go
Comment (rmohr) [reply cluster-provision/gocli/cmd/provision.go:35]: Done.

File: cluster-provision/k8s/check-cluster-up.sh
Comment (rmohr) [reply cluster-provision/k8s/check-cluster-up.sh:6]: Done.

File: cluster-provision/k8s/provision.sh
Comment (rmohr) [reply cluster-provision/k8s/provision.sh:13]: Done.

File: publish.sh
Comment (rmohr) [reply publish.sh:11]: Done.

**Comment by rmohr** [comment]
@qinqon I think most of the issues are addressed now. Could you have another look?

File: cluster-provision/gocli/cmd/provision.go
Comment (qinqon) [reply cluster-provision/gocli/cmd/provision.go]: fair enoug.

File: cluster-provision/gocli/docker/docker.go
Comment (qinqon) [reply cluster-provision/gocli/docker/docker.go:50]: Got it so this is compatible with quay.io.

**Comment by qinqon** [comment]
/lgtm

Let's roll

**Pull request merged**
